gammapy/spectrum/spectrum_analysis.py

```diff
@@ -803,16 +807,32 @@ def _run_sherpa_fit(self):
         ds.notice(thres_lo, thres_hi)
         datastack.set_stat(self.statistic)
         ds.fit()
+        datastack.covar()
+        covar = datastack.get_covar_results()
+        efilter = datastack.get_filter()
+
+        # First go on calculation flux points following
```

Comment:
Flux point computation is a big project on it's own, with some existing code in Gammapy.
For now, can you split this out into a separate function or method from `_run_sherpa_fit` and leave a TODO that this should be improved?